### PR TITLE
feat: execute all tests with lerna to test all packages

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Run commands that need to be run on all packages with Lerna, and use `npm` for the rest.
Lerna can execute all package tests found in the repo
https://github.com/lerna/lerna/tree/main/commands/run#readme

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -226,9 +232,11 @@ jobs:
           name: Update local .npmrc file (Windows)
           command: echo "//registry.npmjs.org/:_authToken=$env:NPM_TOKEN" >> .npmrc
       - run:
-          name: Install dependencies (Windows)
-          command: npm install --ignore-scripts # ignore-scripts required because of postinstall script in snyk-resolve-deps package
-      - build_ts
+          name: Install dependencies @snyk/snyk
+          command: npm install
+      - run:
```
Comment (JackuB) [inline .circleci/config.yml:237]: Doesn't this happen automatically with npm@7? I tried adding `axios` as a local devDep to the protect package and it got installed after `npm i`

Screen Shot 2021-02-18 at 16 49 23

Comment (lili2311) [reply .circleci/config.yml]: removed, you are right. Forgot about npm 7 😅

**Comment by lili2311** [comment]
@JackuB please 👀

File: test/analytics.spec.ts [hunk 2]
```diff
@@ -128,7 +128,7 @@ describe('Analytics basic testing', () => {
         'integrationEnvironmentVersion',
       ].sort(),
     );
-  });
+  }, 10000);
```
Comment (lili2311) [inline test/analytics.spec.ts:131]: these were super flaky and not completing in a default given time

File: .circleci/config.yml [hunk 3]
```diff
@@ -241,25 +247,25 @@ jobs:
           steps:
             - run:
                 name: Run "Root" tap tests
-                command: npm run test:test
+                command: lerna run test:test --stream
```
Comment (JackuB) [inline .circleci/config.yml:250]: So is the plan that each package will have a task called `test:test` and `test:jest`? I think it's just a limitation of the current CLI that we have tests split like this.

How about removing the `test` script from the root package.json (as we can't/don't use it now) and just have a `lerna run test` that will test packages? We might even have a single container dedicated for that now

**Pull request closed without merging**